src/radix_connect/app_link_parser.rs

```diff
@@ -0,0 +1,154 @@
+use crate::prelude::*;
+use url::form_urlencoded;
+use url::Url;
+
+use super::{DappRequest, LinkRequest, MobileConnectRequest};
+
+#[uniffi::export]
+pub fn parse_request_url(url: String) -> Result<MobileConnectRequest> {
+    let connect_url = Url::parse(CONNECT_URL).unwrap();
+    match Url::parse(url.as_str()) {
+        Ok(parsed_url) => {
+            if parsed_url.host_str() != connect_url.host_str()
```

Comment:
Better to avoid "pyramid of doom", by returning early, like so

```rust
   let parsed_url = Url::try_from(&url)
            .map_err(|_| CommonError::RadixConnectMobileInvalidRequestUrl{ bad_value: url })?;
```

Maybe best if you create a pub global utility function for that, and use it here and in `Gateway`. **Notice the `?` in the end, it will return from the whole function if err**, lets for the sake of this review call the function `parse_url`, so:

Somewhere, probably in `src/core/utils` a new file perhaps...? or in `string_utils`:

```rust
pub fn parse_url(s: impl AsRef<str>) -> Result<Url> {
     Url::try_from(url.as_str())
            .map_err(|_| CommonError::InvalidURL { bad_value: url })
}
```

and then here:

```rust
let parsed_url = parse_url(url).map_err(|_| RadixConnectMobileInvalidRequestUrl { bad_value: url })?;
...
```

Reply:
and of course write unit tests for `parse_url` function in that file

Reply:
@jakub-rdx sorry, I meant `as_ref()` ofc :)

Reply:
declared function as 
`pub fn parse_url(s: impl AsRef<str>) -> Result<Url, CommonError> {
    Url::try_from(s.as_ref()).map_err(|_| CommonError::InvalidURL {
        bad_value: s.as_ref().to_owned(),
    })
}` and using it like 
`let parsed_url = parse_url(url).map_err(|_| RadixConnectMobileInvalidRequestUrl { bad_value: url })?;`
produces compiler error